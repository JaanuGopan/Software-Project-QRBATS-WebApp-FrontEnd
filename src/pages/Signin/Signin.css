* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
}

.mainBackground {
    background: url("../../assets/Images/back.jpg") center/cover no-repeat;
    font-family: 'Poppins', sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    display: flex;
    justify-content: center;
    align-items: center;
    min-height: 100vh;
    min-width: 300px;
}

.ComponentBackground {
    position: relative;
    width: 50%;
    border: 4px solid transparent;
    border-top: 6px solid #0077C0;
    border-radius: 10px;
    padding: 20px;
    background-color: rgba(250, 250, 250, 0.97);
    box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1);
    min-width: 250px;
}

.Heading {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: baseline;
    align-items: center;
    padding: 2%;
}

.back-button {
    background: #0077C0;
    border: none;
    padding: 10px;
    cursor: pointer;
    outline: none;
    margin-right: 10px;
    border-radius: 50%;
    /* Ensure the button is rounded */
    transition: background-color 0.3s ease;
    /* Add transition for smooth hover effect */
    display: flex;
    align-items: center;
    justify-content: center;
}

.back-button:hover {
    background: rgba(92, 92, 165, 0.8);
    /* Change the background color on hover */
}

.back-button .icon {
    color: #fff;
    /* Set the color of the icon to white */
}

.content {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    align-items: center;
    justify-content: center;
}

.Designer {
    width: 200px;
    height: auto;
    border-radius: 5px 5px 5px 5px;
}

.head1 {
    text-align: center;
    color: #0063A0;
    font-weight: bold;
    font-size: x-large;
    padding: 1%;
    margin: 0%;
}

.bodyContent {
    width: 35vw;
    height: 30vh;
    max-height: 50%;
    max-width: 50%;
    min-width: 200px;
    display: flex;
    flex-direction: row;
    align-items: center;
}

.login-container {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-top: 20px;
    position: relative;
}

.image-container {
    flex: 1 1;
    display: flex;
    flex-direction: column;
    align-items: center;
}

.form-container {
    flex: 1;
}

.form-group {
    margin-bottom: 15px;
    position: relative;
    display: flex;
    flex-direction: row;
    align-items: center;
}

label {
    display: block;
    margin-bottom: 5px;
}

button {
    padding: 10px 20px;
    background-color: #0077C0;
    color: #fff;
    border: none;
    border-radius: 5px;
    cursor: pointer;
}

a {
    color: #0077C0;
    text-decoration: none;
}

button:hover,
a:hover {
    background-color: transparent;
}

.signin-input-with-icon {
    position: relative;
    display: flex;
    flex-direction: row;
    align-items: center;
    background-color: white;
    border: none;
    margin: 10px 0px;
    border-radius: 5px;
    border-bottom:3px solid #0077C0;
    padding-left: 10px;
}

.input-icon {
    left: 10px;
    top: calc(50% - 0.5em);
    color: #aaa;
}

input[type="text"]::placeholder,
input[type="password"]::placeholder {
    color: #aaa;
}

.signin-input-field {
    padding: 10px;
    border: 0px;
    background-color: transparent;
    box-shadow: none;
    border-bottom: 0px;
    width: 100%;
}

.signin-form-group{
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: stretch;
}

.forgot-password{
    padding: 10px;
}

.forgot-password:hover{
    background-color: transparent;
    text-decoration: underline;
}