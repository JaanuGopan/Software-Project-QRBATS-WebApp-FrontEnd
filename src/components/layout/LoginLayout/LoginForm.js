import React, { useState, useEffect } from "react";
import "../../../pages/Signin/Signin.css";
import { FaUser, FaLock } from "react-icons/fa";
import { useNavigate } from "react-router-dom";
import "bootstrap/dist/css/bootstrap.min.css";
import LoginService from "../../../api/services/LoginService";
import JwtService from "../../../api/services/JwtService";
import { useDispatch, useSelector } from "react-redux";
import { login, selectUser } from "../../../redux/features/userSlice";

const LoginForm = () => {
  const [userName, setUserName] = useState("");
  const [password, setPassword] = useState("");
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const user = useSelector(selectUser);

  // If user is already logged in, redirect to mainNavigation
  useEffect(() => {
    if (user) {
      navigate("/mainNavigation");
    }
  }, [user, navigate]);

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const token = await LoginService.loginUser(userName, password);

      const decodedToken = JwtService.parseJwt(token);
      const loginUserName = decodedToken.sub;
      const lodinUserFirstName = decodedToken.firstName;
      const loginUserLastName = decodedToken.lastName;
      const loginUserEmail = decodedToken.email;
      const loginUserRole = decodedToken.role;
      const loginUserId = decodedToken.userId;
      localStorage.setItem("token", token);

      dispatch(
        login({
          token: token,
          userId: loginUserId,
          userName: loginUserName,
          firstName: lodinUserFirstName,
          lastName: loginUserLastName,
          email: loginUserEmail,
          role: loginUserRole,
          loggedIn: true,
        })
      );

      navigate("/mainNavigation");
    } catch (error) {
      console.error("Login failed", error);
    }
  };

  return (
    <div className="form-container">
      <form onSubmit={handleSubmit}>
        <div className="signin-form-group">
          <div className="signin-input-with-icon">
            <FaUser className="input-icon" />
            <input
              required
              type="text"
              id="userName"
              name="userName"
              placeholder="user name"
              className="signin-input-field"
              value={userName}
              onChange={(e) => setUserName(e.target.value)}
            />
          </div>
          <div className="signin-input-with-icon">
            <FaLock className="input-icon" />
            <input
              required
              type="password"
              id="password"
              name="password"
              placeholder="Password"
              className="signin-input-field"
              value={password}
              onChange={(e) => setPassword(e.target.value)}
            />
          </div>
          <a href="#" className="forgot-password" onClick={() => navigate("")}>
            Forgot Password?
          </a>
          <button type="submit" className="btn btn-primary w-100">
            Login
          </button>
        </div>
      </form>
    </div>
  );
};

export default LoginForm;
