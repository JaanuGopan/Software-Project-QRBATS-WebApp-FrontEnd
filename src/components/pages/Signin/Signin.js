import React from "react";
import "./Signin.css";
import LoginForm from "../../layout/SigninCom/LoginForm";
import Designer from "../../../assets/Images/Designer.jpeg";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCircleArrowLeft } from "@fortawesome/free-solid-svg-icons";

function Signin() {
  return (
    <div className="mainBackground">
      <div className="ComponentBackground">
        <button className="back-button">
          <FontAwesomeIcon icon={faCircleArrowLeft} className="backicon" />
        </button>
        <p className="head1">LOGIN</p>
        <div className="content">
          <div className="image-container">
            <img className="Designer" src={Designer} alt="" />
          </div>
          <div className="bodyContent">
            <LoginForm />
          </div>
        </div>
      </div>
    </div>
  );
}

export default Signin;
