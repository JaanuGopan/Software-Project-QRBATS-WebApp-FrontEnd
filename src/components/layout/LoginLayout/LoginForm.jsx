import React, { useState, useEffect } from 'react';
import '../../../pages/Signin/Signin.css';
import { FaUser, FaLock } from 'react-icons/fa';
import { useNavigate } from 'react-router-dom';
import 'bootstrap/dist/css/bootstrap.min.css';
import { useDispatch, useSelector } from 'react-redux';
import { login, selectUser } from '../../../redux/features/userSlice';
import UserService from '../../../api/services/UserService';
import { ToastContainer, toast } from 'react-toastify';
import { CircularProgress } from '@mui/material';

const LoginForm = ({ handleShowForgotPassword }) => {
  const [userName, setUserName] = useState('');
  const [password, setPassword] = useState('');
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const user = useSelector(selectUser);

  const [processing, setProcessing] = useState(false);

  // If user is already logged in, redirect to mainNavigation
  useEffect(() => {
    if (user) {
      navigate('/mainNavigation');
    }
  }, [user, navigate]);

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      setProcessing(true);
      const response = await UserService.loginUser(userName, password, dispatch);
      if (response.status === 200) {
        toast.success('Successfully LogIn.');
        navigate('/mainNavigation');
      } else if (response.status === 400) {
        toast.error(response.data);
      } else {
        toast.error('Error In LogIn Service.');
      }
    } catch (error) {
      toast.error('Error In LogIn Service.');
      console.error('Login failed', error);
    } finally {
      setProcessing(false);
    }
  };

  return (
    <div className="form-container">
      <form onSubmit={handleSubmit}>
        <div className="signin-form-group">
          <div className="signin-input-with-icon">
            <FaUser className="input-icon" />
            <input
              required
              type="text"
              id="userName"
              name="userName"
              placeholder="user name"
              className="signin-input-field"
              value={userName}
              onChange={(e) => setUserName(e.target.value)}
            />
          </div>
          <div className="signin-input-with-icon">
            <FaLock className="input-icon" />
            <input
              required
              type="password"
              id="password"
              name="password"
              placeholder="Password"
              className="signin-input-field"
              value={password}
              onChange={(e) => setPassword(e.target.value)}
            />
          </div>
          <a className="forgot-password" onClick={handleShowForgotPassword}>
            Forgot Password?
          </a>
          {processing ? (
            <CircularProgress />
          ) : (
            <button id="LoginID" type="submit" className="btn btn-primary w-100">
              Login
            </button>
          )}
        </div>
      </form>
    </div>
  );
};

export default LoginForm;
