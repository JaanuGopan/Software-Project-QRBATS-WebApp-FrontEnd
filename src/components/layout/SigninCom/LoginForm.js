import React, { useState } from "react";
import "../../pages/Signin/Signin.css";
import axios from "axios";
import { FaUser, FaLock } from "react-icons/fa";
import { useNavigate } from "react-router-dom";
import "bootstrap/dist/css/bootstrap.min.css";
import LoginService from "../../../api/services/LoginService";
import JwtService from "../../../api/services/JwtService";
import { useDispatch } from "react-redux";
import { login } from "../../../redux/features/userSlice";

const LoginForm = () => {
  const [userName, setUserName] = useState("");
  const [password, setPassword] = useState("");
  const navigate = useNavigate();

  const dispatch = useDispatch();

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const token = await LoginService.loginUser(userName, password);
      const decodedToken = JwtService.parseJwt(token);
      const loginUserName = decodedToken.sub;
      const lodinUserFirstName = decodedToken.firstName;
      const loginUserLastName = decodedToken.lastName;
      const loginUserEmail = decodedToken.email;
      const loginUserRole = decodedToken.role;
      localStorage.setItem("token", token);

      dispatch(
        login({
          token: token,
          userName: loginUserName,
          firstName: lodinUserFirstName,
          lastName: loginUserLastName,
          email: loginUserEmail,
          role: loginUserRole,
          loggedIn: true,
        })
      );

      navigate("/mainNavigation");
    } catch (error) {
      console.error("Login failed", error);
    }
  };

  return (
    <div className="form-container">
      <form onSubmit={handleSubmit}>
        <div className="form-group">
          <div className="signin-input-with-icon">
            <FaUser className="input-icon" />
            <input
              type="text"
              id="userName"
              name="userName"
              placeholder="user name"
              className="signin-input-field"
              value={userName}
              onChange={(e) => setUserName(e.target.value)}
            />
          </div>
        </div>
        <div className="form-group">
          <div className="signin-input-with-icon">
            <FaLock className="input-icon" />
            <input
              type="password"
              id="password"
              name="password"
              placeholder="Password"
              className="signin-input-field"
              value={password}
              onChange={(e) => setPassword(e.target.value)}
            />
          </div>
        </div>
        <div className="form-group">
          <a href="#" onClick={() => navigate("/forgotpassword")}>
            Forgot Password?
          </a>
        </div>
        <button type="submit" className="btn btn-primary w-100">
          Login
        </button>
      </form>
    </div>
  );
};

export default LoginForm;
