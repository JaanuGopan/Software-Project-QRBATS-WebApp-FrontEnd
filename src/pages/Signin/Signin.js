import React from "react";
import "./Signin.css";
import LoginForm from "../../components/layout/LoginLayout/LoginForm";
import Designer from "../../assets/Images/Designer.jpeg";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCircleArrowLeft } from "@fortawesome/free-solid-svg-icons";
import { useNavigate } from "react-router-dom";

function Signin() {
  const navigate = useNavigate();
  return (
    <div className="mainBackground">
      <div className="ComponentBackground">
        <button onClick={() => navigate("/")} className="back-button">
          <FontAwesomeIcon icon={faCircleArrowLeft} className="backicon" />
        </button>
        <p className="head1">LOGIN</p>
        <div className="content">
          <div className="image-container">
            <img className="Designer" src={Designer} alt="" />
          </div>
          <div className="bodyContent">
            <LoginForm />
          </div>
        </div>
      </div>
    </div>
  );
}

export default Signin;
