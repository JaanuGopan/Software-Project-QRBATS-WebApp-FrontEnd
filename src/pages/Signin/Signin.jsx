import React, { useState } from "react";
import "./Signin.css";
import LoginForm from "../../components/layout/LoginLayout/LoginForm";
import Designer from "../../assets/Images/Designer.jpeg";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCircleArrowLeft } from "@fortawesome/free-solid-svg-icons";
import { useNavigate } from "react-router-dom";
import ForgotPassword from "../ForgotPassword/ForgotPassword";
import OtpVerification from "../ForgotPassword/OtpVerification";
import ResetPassword from "../ForgotPassword/RestPassword";
import { ToastContainer } from "react-toastify";

function Signin() {
  const navigate = useNavigate();
  const [showLogin, setShowLogin] = useState(true);
  const [showForgetPassword, setShowForgotPassword] = useState(false);
  const [showOtpVerification, setShowOtpVerification] = useState(false);
  const [showResetPassword, setShowRestPassword] = useState(false);

  const [title, setTitle] = useState("Login");
  const [email, setEmail] = useState("");

  const handleShowSendOtp = () => {
    setShowLogin(false);
    setShowOtpVerification(false);
    setShowForgotPassword(true);
    setShowRestPassword(false);
    setTitle("Forgot Password");
  };

  const handleCloseSendOtp = () => {
    setShowLogin(true);
    setShowOtpVerification(false);
    setShowForgotPassword(false);
    setShowRestPassword(false);
    setTitle("Login");
  };

  const handleShowOtpVerification = (email) => {
    setEmail(email);
    setShowLogin(false);
    setShowOtpVerification(true);
    setShowForgotPassword(false);
    setShowRestPassword(false);
    setTitle("Otp Verification");
  };
  const handleCloseOtpVerification = () => {
    setShowLogin(false);
    setShowOtpVerification(false);
    setShowForgotPassword(true);
    setShowRestPassword(false);
    setTitle("Forgot Password");
  };
  const handleShowResetPassword = () => {
    setShowLogin(false);
    setShowOtpVerification(false);
    setShowForgotPassword(false);
    setShowRestPassword(true);
    setTitle("Reset Password");
  };
  const handleCloseResetPassword = () => {
    setShowLogin(false);
    setShowOtpVerification(false);
    setShowForgotPassword(true);
    setShowRestPassword(false);
    setTitle("Forgot Password");
  };

  const handleBackButtonAction = () => {
    if (showLogin) {
      navigate("/");
    } else if (showForgetPassword) {
      handleCloseSendOtp();
    } else if (showOtpVerification) {
      handleCloseOtpVerification();
    } else if (showResetPassword) {
      handleCloseResetPassword();
    }
  };

  return (
    <div className="mainBackground">
      <ToastContainer />

      <div className="ComponentBackground">
        <button onClick={handleBackButtonAction} className="back-button">
          <FontAwesomeIcon icon={faCircleArrowLeft} className="backicon" />
        </button>
        <p className="head1">{title}</p>
        <div className="content">
          <div className="image-container">
            <img className="Designer" src={Designer} alt="" />
          </div>
          <div className="bodyContent">
            {showLogin && (
              <LoginForm handleShowForgotPassword={handleShowSendOtp} />
            )}
            {showForgetPassword && (
              <ForgotPassword
                handleCloseForgotPassword={handleCloseSendOtp}
                handleShowOtpVerification={(email) =>
                  handleShowOtpVerification(email)
                }
              />
            )}
            {showOtpVerification && !showForgetPassword && email !== "" && (
              <OtpVerification
                handleCloseOtpVerification={handleCloseOtpVerification}
                handleShowPasswordChange={handleShowResetPassword}
                email={email}
              />
            )}
            {showResetPassword && (
              <ResetPassword
                handleCloseResetPassword={handleCloseResetPassword}
                handleShowLogin={handleCloseSendOtp}
                email={email}
              />
            )}
          </div>
        </div>
      </div>
    </div>
  );
}

export default Signin;
