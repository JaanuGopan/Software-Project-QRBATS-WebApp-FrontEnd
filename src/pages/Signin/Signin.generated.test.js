import renderer from 'react-test-renderer';
import React from "react";

import LoginForm from "../../components/layout/LoginLayout/LoginForm";
import Designer from "../../assets/Images/Designer.jpeg";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCircleArrowLeft } from "@fortawesome/free-solid-svg-icons";
import { useNavigate } from "react-router-dom";
import Signin from "./Signin";

jest.mock("./Signin.css");
jest.mock("../../components/layout/LoginLayout/LoginForm");
jest.mock("../../assets/Images/Designer.jpeg");
jest.mock("@fortawesome/react-fontawesome");
jest.mock("@fortawesome/free-solid-svg-icons");
jest.mock("react-router-dom");

const renderTree = tree => renderer.create(tree);
describe('<Signin>', () => {
  it('should render component', () => {
    expect(renderTree(<Signin 
    />).toJSON()).toMatchSnapshot();
  });
  
});