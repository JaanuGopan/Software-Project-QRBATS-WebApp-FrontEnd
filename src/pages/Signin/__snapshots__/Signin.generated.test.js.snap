// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signin> should render component 1`] = `
<div
  className="mainBackground"
>
  <div
    className="ComponentBackground"
  >
    <button
      className="back-button"
      onClick={[Function]}
    />
    <p
      className="head1"
    >
      LOGIN
    </p>
    <div
      className="content"
    >
      <div
        className="image-container"
      >
        <img
          alt=""
          className="Designer"
          src="Designer.jpeg"
        />
      </div>
      <div
        className="bodyContent"
      />
    </div>
  </div>
</div>
`;
